File: sdk/scala-sdk-protobuf/src/main/scala/kalix/scalasdk/replicatedentity/ReplicatedSetEntity.scala
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.scalasdk.replicatedentity

class ReplicatedSetEntity[E] extends ReplicatedEntity[ReplicatedSet[E]] {

  /**
   * Implement by returning the initial empty replicated data object. This object will be passed into the command
   * handlers.
   *
   * Also known as the "zero" or "neutral" state.
   *
   * The initial data cannot be `null`.
   *
   * @param factory
   *   the factory to create the initial empty replicated data object
   */
  override final def emptyData(factory: ReplicatedDataFactory): ReplicatedSet[E] =
    factory.newReplicatedSet
}

File: sdk/java-sdk-protobuf/src/main/java/kalix/javasdk/replicatedentity/ReplicatedSetEntity.java
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.javasdk.replicatedentity;

public class ReplicatedSetEntity<T> extends ReplicatedEntity<ReplicatedSet<T>> {
  @Override
  public final ReplicatedSet<T> emptyData(ReplicatedDataFactory factory) {
    return factory.newReplicatedSet();
  }
}
